### ISSUE TITLE
cide--get-string-from-file: Wrong type argument

### ISSUE THREAD

**Issue by szikes-adam** [atilaneves/cmake-ide#188]
I have newly installed the cmake-ide. When I run the `M-x cmake-ide-run-cmake RET`, cmake-ide will say:
> cmake-ide [Sat Apr 13 22:02:40 2019]: Making directory ...
> cmake-ide [Sat Apr 13 22:02:40 2019]: Running cmake for src path ... in build path ...
> cmake-ide [Sat Apr 13 22:02:48 2019]: Finished running CMake
> error in process sentinel: cide--get-string-from-file: Wrong type argument: stringp, nil
> error in process sentinel: Wrong type argument: stringp, nil

The compile_commands.json file is generated.

I tried the `M-x cmake-ide-run-cmake RET` with a totally different project but the result was the same.

**Environment:**
Ubuntu
cmake 3.5.1
Emacs 26.1
cmake-ide 20190313.831

**Comment by atilaneves** [comment]
Are either of those open-source projects?

**Comment by buherton** [comment]
Yes, both of them:
https://github.com/bast/gtest-demo
https://github.com/Andersbakken/rtags

Let me know, if I can help somehow.

**Comment by szikes-adam** [comment]
I did some investigation. When the cmake ends, the `cide--locate-cmakelists` will be evaluated based on the **current** `default-directory`. So, if you switch buffer to see what is happening, you will have this issue.

I think my first problem was due to CMakeCache.txt. I built locally first time then I left the CMakeCache.txt in the source directory. See: [link](https://stackoverflow.com/questions/24070260/cmake-putting-build-files-in-source-directory)
